src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +197,17 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      propsMutated = !shallowEqual(inst.props, publicProps);
+      if (propsMutated) {
+        warning(
+          !propsMutated,
+          'Constructor (of %s) may not invoke super(...) with props that ' +
```

Comment:
@jimfb @spicyj Can we have a consensus over the warning message? I personally find both the messages good enough.